File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 11/06/2025, 14:58:36
class Solution {
    public int lengthOfLongestSubstring(String s) {
      int start = 0;
      int end = 0;

      int max = 0;

      HashSet<Character> set = new HashSet<>();


      while(end<s.length()){
        if(!set.contains(s.charAt(end))){
            set.add(s.charAt(end));
            max = Math.max(max, end-start+1);
            end++;
        }
        else{
            set.remove(s.charAt(start));
            start++;
        }
      }

      return max;  
    }
}

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.py
# Last updated: 08/04/2025, 18:13:38
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start=0
        end=0

        hash_set = set()
        maxLength=0

        while(start<=end and end<len(s)):
            if s[end] not in hash_set:
                hash_set.add(s[end])
                maxLength=max(maxLength,end-start+1)
                end+=1
            else:
                hash_set.remove(s[start])
                start+=1

        
        return maxLength

        